prober/serviceimpl.go

```diff
@@ -69,10 +69,16 @@ func (pr *Prober) ListProbes(ctx context.Context, req *pb.ListProbesRequest) (*p
 	defer pr.mu.Unlock()
 
 	resp := &pb.ListProbesResponse{}
-	for name, p := range pr.Probes {
+
+	keys := make([]string, 0)
+	for k, _ := range pr.Probes {
+		keys = append(keys, k)
+	}
+
+	for i := 0; i < len(keys); i++ {
 		resp.Probe = append(resp.Probe, &pb.Probe{
-			Name:   &name,
```

Comment:
Actually the easier and idiomatic solution for this is to use proto.String(name) here. Can you please do that instead. 

proto.String allocates a new string variable with the value "name" and assigns the address of that to the field. 

Below, pr.Probes[name].ProbeDef should work as storage for that pointer won't change.

Reply:
That's a much neater way of doing this, I didn't about that. Thank you!

I've made these changes.